cpp/log/signer.cc

```diff
@@ -59,7 +63,12 @@ std::string Signer::RawSign(const std::string& data) const {
   unsigned int sig_size = EVP_PKEY_size(pkey_.get());
   unsigned char* sig = new unsigned char[sig_size];
 
-  CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
+  if (synchronize_signing_) {
+    std::lock_guard<std::mutex> lock(lock_);
+    CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
```

Comment:
I'm wondering if it's worth sub-classing the Signer class to have a SingleThreadedSigner that will lock before calling EVP_SignFinal (so only the call to EVP_SignFinal would be delegated).

That way the right signer would be created from configuration during initialization, it will not change, and there won't be two flows to check/fix should any bugs be discovered here.